packages/std/src/math/uint256.rs

```diff
@@ -222,71 +238,91 @@ impl From<Uint256> for String {
 
 impl fmt::Display for Uint256 {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        self.0.fmt(f)
+        // The inner type doesn't work as expected with padding, so we
+        // work around that.
+        let unpadded = self.0.to_string();
+
+        f.pad_integral(true, "", &unpadded)
```

Comment:
Nice

Could you pull out this fix in a separate PR and add tests that failed before the fix? Also let's add the same tests to Uint64 and Uint128.

Reply:
Done in https://github.com/CosmWasm/cosmwasm/pull/1069; Thanks